src/Middleware/RequestThrottling/test/RequestQueueTests.cs

```diff
@@ -15,7 +15,7 @@ public async Task LimitsIncomingRequests()
             using var s = new RequestQueue(1);
             Assert.Equal(1, s.Count);
 
-            await s.EnterQueue().OrTimeout();
+            await s.TryEnterQueueAsync().OrTimeout();
```

Comment:
Assert the return value is true for all the current calls to TryEnterQueueAsync.